### PR TITLE
docs cleanup

### PR DESCRIPTION
**Pull request by macovsky**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,39 +1,30 @@
-# PG2: A *Fast* PostgreSQL Driver For Clojure
+# PG2: *Fast* PostgreSQL driver for Clojure
 
-[pg]: https://github.com/igrishaev/pg
+`PG2` is a JDBC-free PostgreSQL client library.
 
-PG2 is a client library for PostgreSQL server. It succeeds [PG(one)][pg] -- my
```
Comment (macovsky) [inline README.md:5]: sorry I removed it since it seems to be a bit off-topic, IMO it's important on https://github.com/igrishaev/pg only

or maybe it can be moved to the bottom as a history, it's your call ofc

File: README.md [hunk 2]
```diff
@@ -1,39 +1,30 @@
-# PG2: A *Fast* PostgreSQL Driver For Clojure
+# PG2: *Fast* PostgreSQL driver for Clojure
 
-[pg]: https://github.com/igrishaev/pg
+`PG2` is a JDBC-free PostgreSQL client library.
```
Comment (macovsky) [inline README.md:3]: maybe you can come up with some description which you use in both title here and repo title

File: docs/json.md [hunk 3]
```diff
@@ -65,39 +61,11 @@ fetch the new row by id:
...
-(pgh/insert-one conn
-                :test_json
-                {:data [:param :data]}
-                {:honey {:params {:data {:some [:json {:map [1 2 3]}]}}}})
-~~~
-
-For details, see the [HoneySQL Integration](docs/honeysql.md) section.
+Again JSON returns as a Clojure map with no wrappers.
 
-PG2 supports only Clojure maps when encoding values into JSON. Vectors and other
```
Comment (macovsky) [inline docs/json.md:96]: is this still true? because on top of this file you said both vectors and maps

if it is I'll return this ofc

File: docs/transaction.md [hunk 4]
```diff
@@ -157,70 +157,6 @@ refer to the [official documentation][transaction-iso] for more information.
 
 ## Nested Transactions
 
-Nested transactions happen when you put one `with-tx` inside another, for
```
Comment (macovsky) [inline docs/transaction.md:160]: sorry I removed this passage as it seems to be outdated: pg2 works fine with nested `with-tx` now

File: docs/connecting.md [hunk 5]
```diff
@@ -74,54 +72,38 @@ are plain strings:
...
-
-2. It prevents the `:read-only?` flag from overriding in the `with-tx`
-   macro. Say, even if the macro is called like this:
-
-~~~clojure
-(pg/with-tx [conn {:read-only? false}] ;; try to mute the global :read-only? flag
-  (pg/query conn "delete from students"))
-~~~
-
-The transaction will be in `READ ONLY` mode anyway.
```
Comment (igrishaev) [inline docs/connecting.md:127]: Personally I wouldn't like to drop out this fragment. Can you get it back please? The phrasing you proposed is too short and I'm afraid the context will be lost.

File: docs/arrays.md [hunk 6]
```diff
@@ -1,23 +1,23 @@
 # Arrays support
 
-In JDBC, arrays have always been a pain. Every time you're about to pass an
```
Comment (igrishaev) [inline docs/arrays.md:3]: Here and below: I don't think these commas should be removed. The "In JDBC" clause in the introductory, and thus must be separated with comma. That's the number 2 point from this guide:

https://www.brandeis.edu/writing-program/resources/students/academic/style-grammar/comma-rules.html

But can you change my mind?

File: README.md [hunk 7]
```diff
@@ -1,39 +1,30 @@
-# PG2: A *Fast* PostgreSQL Driver For Clojure
```
Comment (igrishaev) [inline README.md:1]: Why did you remove A? It relates to the "Driver" noun which is mentioned the first time.

File: README.md [hunk 8]
```diff
@@ -1,39 +1,30 @@
...
+**It's fast**: [benchmarks](/docs/benchmarks.md) prove up to 3 times performance boost compared to
+`next.jdbc`. Simple HTTP application reading from the database and
+serving JSON handles 2 times more RPS.
 
-**It's fast.** Benchmarks prove up to 3 times performance boost compared to
-Next.JDBC. A simple HTTP application which reads data from the database and
-responds with JSON handles 2 times more RPS. For details, see the "benchmarks"
-below.
-
-**It's written in Java** with a Clojure layer on top of it. Unfortunately,
```
Comment (igrishaev) [inline README.md:14]: Unfortunately,

comma is needed here

File: docs/arrays.md (hunk 6 above)
Comment (igrishaev) [inline docs/arrays.md:3]: the `an` article must be kept

File: docs/arrays.md [hunk 9]
```diff
@@ -1,23 +1,23 @@
...
+but `uuid`, `numeric`, `timestamp(tz)`, `json(b)`, and more.
 
-Arrays might have more than one dimension. Nothing prevents you from having a 3D
+Arrays can have more than one dimension. Nothing prevents you from having a 3D
 array of integers like `cube::int[][][]`, and it becomes a nested vector when
-fetched by PG2.
+fetched by `PG2`.
 
-*A technical note: PG2 supports both encoding and decoding of arrays in both
+*Technical note: `PG2` supports both encoding and decoding of arrays in both
```
Comment (igrishaev) [inline docs/arrays.md:20]: A article must be preserved, I think

File: docs/arrays.md [hunk 10]
```diff
@@ -34,7 +34,7 @@ Insert a simple item:
             {:params [["one" "two" "three"]]})
 ~~~
 
-In arrays, some elements might be NULL:
```
Comment (igrishaev) [inline docs/arrays.md:37]: Comma is needed (try this sentence in Grammarly or similar service)

File: docs/arrays.md [hunk 11]
```diff
@@ -51,7 +51,7 @@ Now let's check what we've got so far:
  {:id 2 :text_arr ["foo" nil "bar"]}]
 ~~~
 
-Postgres supports plenty of operators for arrays. Say, the `&&` one checks if
+Postgres supports plenty of operators for arrays. Say `&&` checks if
```
Comment (igrishaev) [inline docs/arrays.md:54]: Comma is needed

File: docs/arrays.md [hunk 12]
```diff
@@ -92,8 +92,8 @@ Here is how you insert a matrix:
 {:inserted 1}
 ~~~
 
-Pay attention: each number can be NULL but you cannot have NULL for an entire
-sub-array. This will trigger an error response from Postgres.
+Pay attention: each number can be NULL but you can't have NULL for an entire
```
Comment (igrishaev) [inline docs/arrays.md:95]: I'd prefer `cannot` rather than can't (and you know why, all know)

File: docs/connecting.md [hunk 13]
```diff
@@ -17,24 +16,24 @@ function:
...
-The `:host`, `:port`, and `:password` config fields have default values and
-might be skipped (the password is an empty string by default). Only the `:user`
-and `:database` fields are required when connecting. See the list of possible
-fields and their values in a separate section.
+`:host`, `:port`, and `:password` have default values and
+can be skipped (password is an empty string by default). Only `:user`
+and `:database` are required. See the list of possible
+options below.
 
-To close a connection, pass it into the `close` function:
```
Comment (igrishaev) [inline docs/connecting.md:27]: comma is needed

File: docs/connecting.md [hunk 14]
```diff
@@ -17,24 +16,24 @@ function:
...
+To close the connection call `close` with it:
```
Comment (igrishaev) [inline docs/connecting.md:26]: comma is needed

File: docs/connection-state.md [hunk 15]
```diff
@@ -1,7 +1,6 @@
 # Connection state
 
-There are some function to track the connection state. In Postgres, the state of
-a connection might be one of these:
+In Postgres the connection state might be one of these:
```
Comment (igrishaev) [inline docs/connection-state.md:3]: comma is needed

File: docs/connection-state.md [hunk 16]
```diff
@@ -56,9 +59,9 @@ The connection is in the error state now:
 true
 ~~~
 
-When state is error, the connection doesn't accept any new queries. Each of them
+When state is error the connection doesn't accept any new queries. Each of them
```
Comment (igrishaev) [inline docs/connection-state.md:62]: comma

File: docs/connecting.md [hunk 17]
```diff
@@ -74,54 +72,38 @@ are plain strings:
...
+| `:so-tcp-no-delay?`    | bool         | `true`               | Socket TcpNoDelay value                                                                   |
+| `:so-timeout`          | integer      | `15.000`             | Socket timeout value (in ms)                                                              |
+| `:so-recv-buf-size`    | integer      | `0xFFFF`             | Socket receive buffer size                                                                |
+| `:so-send-buf-size`    | integer      | `0xFFFF`             | Socket send buffer size                                                                   |
+| `:cancel-timeout-ms`   | integer      | `5000`               | Default value for the `with-timeout` macro (in ms)                                        |
+| `:protocol-version`    | integer      | `196608`             | Postgres protocol version                                                                 |
+| `:object-mapper`       | ObjectMapper | `JSON.defaultMapper` | Instance of `ObjectMapper` for custom JSON processing (see [JSON support](/docs/json.md)) |
+
+### read-only mode
+
```
Comment (igrishaev) [inline docs/connecting.md:108]: (duplicate): keep the origin block of code with examples here
